src/Microsoft.DotNet.Build.Tasks/PackageFiles/ApiCompat.targets

```diff
@@ -0,0 +1,137 @@
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <UsingTask TaskName="PrereleaseResolveNuGetPackageAssets" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="LocatePreviousContract" AssemblyFile="$(BuildToolsTaskDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  
+  <PropertyGroup>
+    <ApiCompatBaseline Condition="'$(targetGroup)'!=''">$(MSBuildProjectDirectory)\ApiCompatBaseline.$(targetGroup).txt</ApiCompatBaseline>
+    <ApiCompatBaseline Condition="'$(targetGroup)'==''">$(MSBuildProjectDirectory)\ApiCompatBaseline.txt</ApiCompatBaseline>
+    <ReferenceProjectDirectory>$(MSBuildProjectDirectory)\..\ref</ReferenceProjectDirectory>
+    <ReferenceProject>$(ReferenceProjectDirectory)\$(MSBuildProjectFile)</ReferenceProject>
+    <ReferenceOutputPath>$(ReferenceAssemblyOutputPath)$(AssemblyName)/$(AssemblyVersion)</ReferenceOutputPath>
+    <ReferenceAssembly>$(ReferenceOutputPath)\$(AssemblyName).dll</ReferenceAssembly>
+    <ReferenceProjectJson Condition="'$(ReferenceProjectJson)'=='' and Exists('$(ReferenceProjectDirectory)/project.json')">$(ReferenceProjectDirectory)/project.json</ReferenceProjectJson>
+    <ReferenceProjectLockJson Condition="Exists('$(ReferenceProjectJson)') and '$(ReferenceProjectLockJson)'==''">$(ReferenceProjectDirectory)/project.lock.json</ReferenceProjectLockJson>    
+  </PropertyGroup>
+  
+  <PropertyGroup>
+    <RunApiCompatForSrc Condition="$(MSBuildProjectDirectory.EndsWith('src'))">true</RunApiCompatForSrc>
+    <RunApiCompatForRef Condition="$(MSBuildProjectDirectory.EndsWith('ref'))">true</RunApiCompatForRef>
+    <RunApiCompat Condition="'$(RunApiCompat)'==''">true</RunApiCompat>
```

Comment:
i also ran builds with and without ApiCompat to gather data. It seems to be adding roughly around a minute to the previous build time on corefx. It takes about 7-8 minutes without apicompat and 8-9 with it. I will give you more exact numbers if you want.